Tools/TestWebKitAPI/Tests/WPEPlatform/TestSettings.cpp

```diff
@@ -0,0 +1,231 @@
+    GRefPtr<GVariant> anotherValue = g_variant_new_int32(10000);
+    GUniqueOutPtr<GError> error;
+    gboolean ret;
+
+    g_key_file_set_integer(keyFile.get(), "wpe-platform", "a-key", 42);
+
+    // Loading a key that has not been registered.
+    ret = wpe_settings_load_from_keyfile(settings.get(), keyFile.get(), &error.outPtr());
+    g_assert_error(error.get(), WPE_SETTINGS_ERROR, WPE_SETTINGS_ERROR_NOT_REGISTERED);
+    g_assert_false(ret);
+
+    // Loading a registered key.
+    ret = wpe_settings_register(settings.get(), "/wpe-platform/a-key", G_VARIANT_TYPE_INT32, someValue.get(), &error.outPtr());
+    g_assert_no_error(error.get());
+    g_assert_true(ret);
+    ret = wpe_settings_load_from_keyfile(settings.get(), keyFile.get(), &error.outPtr());
+    g_assert_no_error(error.get());
+    g_assert_true(ret);
+    g_assert_cmpvariant(wpe_settings_get_value(settings.get(), "/wpe-platform/a-key", nullptr), someValue.get());
+
+    // Loading again overwrites values.
+    g_key_file_set_integer(keyFile.get(), "wpe-platform", "a-key", 10000);
+    ret = wpe_settings_load_from_keyfile(settings.get(), keyFile.get(), nullptr);
+    g_assert_true(ret);
+    g_assert_cmpvariant(wpe_settings_get_value(settings.get(), "/wpe-platform/a-key", nullptr), anotherValue.get());
+
+    // Loading a key with a different type.
+    g_key_file_set_string(keyFile.get(), "wpe-platform", "a-key", "'not-a-number'");
+    ret = wpe_settings_load_from_keyfile(settings.get(), keyFile.get(), &error.outPtr());
+    g_assert_error(error.get(), WPE_SETTINGS_ERROR, WPE_SETTINGS_ERROR_INVALID_VALUE);
+    g_assert_false(ret);
+}
+
+static void testSettingsChangedSignal(Test*, gconstpointer)
+{
+    GRefPtr<WPESettings> settings = adoptGRef(WPE_SETTINGS(g_object_new(WPE_TYPE_SETTINGS, nullptr)));
+    GRefPtr<GVariant> someValue = g_variant_new_int32(42);
+    GRefPtr<GVariant> anotherValue = g_variant_new_int32(10000);
+
+    wpe_settings_register(settings.get(), "/wpe-platform/a-key", G_VARIANT_TYPE_INT32, someValue.get(), nullptr);
+    wpe_settings_register(settings.get(), "/wpe-platform/another-key", G_VARIANT_TYPE_INT32, anotherValue.get(), nullptr);
+
+    // Changing a specific key.
+    int count = 0;
+    int handler = g_signal_connect(settings.get(), "changed::/wpe-platform/a-key", G_CALLBACK(+[](WPESettings*, const char* key, GVariant* value, int* count) {
+        g_assert_cmpstr(key, ==, "/wpe-platform/a-key");
+        g_assert_true(g_variant_type_equal(g_variant_get_type(value), G_VARIANT_TYPE_INT32));
+        g_assert_cmpint(g_variant_get_int32(value), ==, 42);
+        (*count)++;
+    }), &count);
+    wpe_settings_set_value(settings.get(), "/wpe-platform/a-key", someValue.get(), FALSE, nullptr);
+    wpe_settings_set_value(settings.get(), "/wpe-platform/another-key", anotherValue.get(), FALSE, nullptr);
+    g_signal_handler_disconnect(settings.get(), handler);
+    g_assert_cmpint(count, ==, 1);
```

Comment:
Why is it emitted in this case? The same value is set, no? hmm, I guess we don't compare to the default value?

Reply:
Changed to only emit when new value doesn't equal old.